### ISSUE TITLE
Email not sent on automatic affiliate registrations

### ISSUE THREAD

**Issue by sumobi** [affiliatewp/AffiliateWP#727]
`Automatically register new users as affiliates?` enabled
`Notify site admins of new affiliate registrations?` enabled 

If a user is automatically registered as an affiliate, the admin does not receive an email. The admin email function is currently only fired on the `affwp_register_user` hook, which only exists inside the `register_user` function.

Likewise, an affiliate will not receive a pending affiliate registration email if `Require that site admins approve affiliates before they can begin earning referrals?` is enabled. They will however receive the application accepted email if the setting is not active.

I think the admin email and the application pending email should be sent when users are automatically added as affiliates. Do you agree @pippinsplugins ?

**Comment by pippinsplugins** [comment]
Yes, I agree.

**Comment by codelion7** [comment]
I second that. I think it would be nice to have some sort of hook that fires during the auto affiliate registration process. Preferably one that passes the same parameters as the affwp_register_user hook does.

With AffiliateWP MLM, this has been my challenge in creating a way to connect Sub Affiliates to Parent Affiliates during the auto affiliate registration process, instead of being limited to the Affiliate Registration Form only.

Do you think that adding a new hook is the best way to accomplish this?

**Comment by pippinsplugins** [comment]
New hook sounds fine to me!

On Fri, Jun 19, 2015 at 5:37 PM, Christian Freeman <notifications@github.com

**Comment by pippinsplugins** [comment]
@sumobi Could you give this a quick test? Should be complete.

**Comment by sumobi** [comment]
@fjarrett @pippinsplugins checked out release/1.7, the emails are not sending when a user is automatically added as an affiliate

**Comment by fjarrett** [comment]
@sumobi You need to check out this branch, the PR is not merged into 1.7 yet

**Comment by fjarrett** [comment]
@sumobi Hmmm no you are right, it was merged. Still, does it work when this branch is checked out?

**Comment by fjarrett** [comment]
@pippinsplugins This is strange. Everything works for me from the `Issue/727` branch, but not from the `release/1.7` branch. Just want to confirm that this was indeed merged, right?

**Comment by fjarrett** [comment]
@sumobi @pippinsplugins Ah, I think I found the problem here, remember what we added in #659?

Make sure this setting is unchecked :wink: 

![screen shot 2015-07-21 at 3 00 47 pm](https://cloud.githubusercontent.com/assets/522158/8811142/8d3278ac-2fb9-11e5-970e-54a46e0bb6c1.png)

Works for me.

**Comment by pippinsplugins** [comment]
Heh, I bet you're right :D

**Comment by sumobi** [comment]
@fjarrett @pippinsplugins I can confirm the emails are now sent. Weird though, never had that new checkbox option enabled  :)

**Comment by johndjchan** [comment]
Hi there, I liked your page and I bought the plugin. The USD199 version. I spoke with jason from s2memeber and understand that affiliatewp also using the wp_user database, so that now, when my user register using UserPro or UserUltra , they can be seen in s2member and also affiliateWP I have went to AffiliateWP and set it in MISC -> "Automatically register new users as affiliates?" and also, UNticked " Require that site admins approve affiliates before they can begin earning referrals? " so, when now my user register in frontend userultra, it working well showing it as affiliate, BUT, the problem is, its now a pending status, and i had to go in and tick the box myself to approve it. can you pls let me know how to make it auto approve? thanks. sorry had to ask it here, but i dont get a reply from your support email and its urgent... Peace, and btw its a great plugin, i bought this instead of iDevaffiliate =)

in short, the auto approve of the affiliate account is not working :(

**Comment by pippinsplugins** [comment]
Hi @johndjchan, I saw you submitted a ticket via our support system (and also from the support page of pippinsplugins.com).

We will address your ticket as soon as we are able. Note: Gthub (this site) is only for development tracking and not support. Also, please only submit one ticket. There's no need to submit the same ticket 3 times. We'll see it :)

**Comment by johndjchan** [comment]
Thanks

i bought the plugin but some critical part is not working, therefore i
asked couple of times
sry mate.

thanks

amanda/john

2015-09-28 8:22 GMT+08:00 Pippin Williamson <notifications@github.com>:
